cmdlets/sqlserver/New-MiLink.ps1

```diff
@@ -0,0 +1,318 @@
+        if ($recoveryModel -ne 1) {
+            Write-Verbose "Set recovery model to FULL for database $DatabaseName [started]"
+            Invoke-SqlCmd -Query "ALTER DATABASE [$DatabaseName] SET RECOVERY FULL" -ServerInstance $SqlInstance      
+            Write-Verbose "Set recovery model to FULL for database $DatabaseName [completed]"
+        }
+
+        $backupHistory = Get-SqlBackupHistory -ServerInstance $SqlInstance -DatabaseName $DatabaseName
+        if (!$backupHistory) {
+            $backupDiskPath = Read-Host -Prompt "Creating database backup, enter path"
+            Write-Verbose "Execute backup for all databases you want to replicate [started]"
+            Invoke-SqlCmd -Query "BACKUP DATABASE [$DatabaseName] TO DISK = N'$backupDiskPath'" -ServerInstance $SqlInstance
+            Write-Verbose "Execute backup for all databases you want to replicate [completed]"
+            Get-SqlBackupHistory -ServerInstance $SqlInstance -DatabaseName $DatabaseName
+        }
+
+        Write-Verbose "Get MI info ($ManagedInstanceName) [started]"
+        $instance = Get-AzSqlInstance -ResourceGroupName $ResourceGroupName -Name $ManagedInstanceName
+        $miFQDN = $instance.FullyQualifiedDomainName       
+        $SqlMiInstanceConnectionString = "tcp://${miFQDN}:5022;Server=[$ManagedInstanceName]"
+        Write-Verbose "Get MI info ($ManagedInstanceName) [completed]"
+
+        # Figure out if we're on a VM https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=windows#example-scenarios-for-usage
+        Write-Verbose "Get VM info [started]"
+        $vmMetadata = Invoke-RestMethod -Headers @{"Metadata" = "true" } -Method GET -NoProxy -Uri "http://169.254.169.254/metadata/instance?api-version=2021-02-01" 
+        $vmName = $vmMetadata.psobject.properties['compute'].value.name
+        $vmRgName = $vmMetadata.psobject.properties['compute'].value.resourceGroupName
+        $vm = Get-AzVm -ResourceGroupName $vmRgName -Name $vmName
+        $nic = $vm.NetworkProfile.NetworkInterfaces[0] # TODO we can have more?
+        $networkinterface = ($nic.id -split '/')[-1]
+        $nicdetails = Get-AzNetworkInterface -Name $networkinterface
+        # Grab the private ip of the server
+        $srcIP = $nicdetails.IpConfigurations[0].PrivateIpAddress
+        $sourceEndpoint = "tcp://${srcip}:5022"
+        Write-Verbose "Get VM info [completed]"
+
+        # TODO:
+        # check if managed instance has <100 DBs
+        # Check if MI has enough storage
+
+        # TODO: If we don't have DBM endpoint, we have to create it
+        # we can either create a new certificate for that purpose, or pick an existing one if it meets certain conditions such as:
+        # - "select pvt_key_encryption_type from sys.certificates where name = N'$boxCertName'" must be MK
+        # TODO: can we trust sp_get_endpoint_certificate ?
```

Comment:
No reason it shouldn't work. Documentation explicitly mentions it in relationship with Managed Instance Link.

https://docs.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sp-get-endpoint-certificate-transact-sql?view=sql-server-ver16

Reply:
I wasn't sure whether the certificate returned by this SP is guaranteed to have its private key encrypted by MK (when ran in on-prem, not MI). i'll try to test for such case and see if it complains